### PR TITLE
SI-7328 Bail out of names/defaults when args are error typed

### PR DESCRIPTION
**Pull request by retronym**
To avoid a crasher later on with a null type inside a
sequence argument.

Review by @lrytz

### CODE REVIEW

File: src/compiler/scala/tools/nsc/typechecker/NamesDefaults.scala [hunk 1]
```diff
@@ -361,6 +360,7 @@ trait NamesDefaults { self: Analyzer =>
               context.namedApplyBlockInfo =
                 Some((block, NamedApplyInfo(qual, targs, vargss :+ refArgs, blockTyper)))
               block
+            case _ => tree
```
Comment (xeno-by) [inline src/compiler/scala/tools/nsc/typechecker/NamesDefaults.scala:363]: Hmm I wonder why this clause was missing from this match before. Does it prevent some sort of a crash for the case of an elaborate erroneous tree?

Comment (retronym) [inline src/compiler/scala/tools/nsc/typechecker/NamesDefaults.scala:363]: I merged the enclosing `if` with the `match`. In the test case, the `typedApp` was not erroneous; the repeated argument _was_ erroneous and contained a subtree with a `null` type. That type was used by `NamesDefaults`, passed to `setInfo`, and hence the crash.

**Comment by lrytz** [comment]
LGTM

**Comment by adriaanm** [comment]
SI-7238, actually -- I love it when I catch stuff like this the very second after I hit merge.

**Comment by retronym** [comment]
Darn... I'm chronically afflicted with JIRA lysdexia. Perhaps the build bot should comment here with links and titles of referenced bugs.

**Comment by adriaanm** [comment]
Absolutely agree. I'm kind of done with built botting for a while, though.
Next time I work on it will hopefully be to turn it into a play app that exposes a service hook to github and similarly for jenkins. It must be reactive. None of this dirty polling business.

**Pull request merged**